xdsl/dialects/linalg.py

```diff
@@ -416,12 +416,55 @@ def __init__(
         )
 
 
+@irdl_op_definition
+class FillOp(IRDLOperation):
+    """
+    Fills the output tensor with the given value.
+
+    See https://mlir.llvm.org/docs/Dialects/Linalg/#linalgfill-linalgfillop
+    """
+
+    name = "linalg.fill"
+
+    inputs = var_operand_def()
+    outputs = var_operand_def(AnyShapedType())
+
+    res = var_result_def(AnyTensorType)
+
+    assembly_format = (
+        "`ins` `(` $inputs `:` type($inputs) `)` ` ` "
+        "`outs` `(` $outputs `:` type($outputs) `)` `->` type($res) attr-dict"
+    )
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True), ParsePropInAttrDict()]
+
+    def __init__(
+        self,
+        inputs: Sequence[SSAValue],
+        outputs: Sequence[SSAValue] = (),
+        res: Sequence[Attribute] | None = None,
+    ):
+        if res is None:
+            result_types = tuple(output.type for output in outputs)
+        else:
+            result_types = res
+
+        super().__init__(
+            operands=(inputs, outputs),
+            result_types=result_types,
+        )
+        # Numeric casting is performed on the value operand, promoting it to the same data type as the output.
+        if outputs:
+            self.inputs = [cast(outputs[0].type, operand) for operand in inputs]
```

Comment:
this is not needed, the casting is at runtime

Reply:
The meaning is that you can fill a tensor of f64 with inputs of type f32, these types don't need to match. I don't know what kind of validation MLIR does on compatibility of types.

Reply:
ahh that makes sense, thanks